src/ansys/grantami/recordlists/_connection.py

```diff
@@ -530,6 +578,94 @@ def unsubscribe_from_list(self, record_list: RecordList) -> None:
+        self._record_versions_api = api.RecordsRecordVersionsApi(client)
+        self._db_schema_api = api.SchemaDatabasesApi(client)
+        self._db_map: Dict[str, str] = {}
+        self._read_mode = read_mode
+
+    def get_resolvable_items(self, all_items: List[RecordListItem]) -> List[RecordListItem]:
+        """Test all items to see if they can be resolved as the current user.
+
+        Uses concurrent.futures to handle threading, since openapi-common is currently based on
+        requests which limits asyncio options.
+
+        Parameters
+        ----------
+        all_items
+            The items for which to check resolvability.
+
+        Returns
+        -------
+        resolvable_items
+            The items which could be resolved on the server.
+        """
+        self._db_map = self._get_db_map()
+        with concurrent.futures.ThreadPoolExecutor(max_workers=self._max_requests) as executor:
+            resolvable_test_futures = {
+                executor.submit(self._is_item_resolvable, i): i for i in all_items
+            }
+            resolvable_items = [
+                resolvable_test_futures[f]
+                for f in concurrent.futures.as_completed(resolvable_test_futures)
+                if f.result()
+            ]
+        return resolvable_items
+
+    def _get_db_map(self) -> Dict[str, str]:
+        dbs = self._db_schema_api.get_all_databases()
+        return {db.guid: db.key for db in dbs.databases}
+
+    def _is_item_resolvable(self, item: RecordListItem) -> bool:
+        """Test if a specific item is resolvable.
+
+        If the item has a record version and record guid, attempt to resolve the record version
+        directly.
+
+        If the item has a record version number only, check that the version number is available
+        to the user running the record.
+
+        If there is no version at all, check that the history can be resolved.
+
+        Returns
+        -------
+        bool
+            True if the item can be resolved, False otherwise.
+        """
+        if item.database_guid not in self._db_map:
+            return False
+        try:
+            if item.record_version is not None and item.record_guid is not None:
+                self._record_versions_api.get_record_version(
+                    database_key=self._db_map[item.database_guid],
+                    table_guid=item.table_guid,
+                    record_history_guid=item.record_history_guid,
+                    record_version_guid=item.record_guid,
+                    mode="read" if self._read_mode else None,
+                )
+            else:
+                history_info = self._record_histories_api.get_record_history(
+                    database_key=self._db_map[item.database_guid],
+                    record_history_guid=item.record_history_guid,
+                    mode="read" if self._read_mode else None,
+                )
+                if item.record_version is not None:
```

Comment:
This is weird to me. From what I've seen, a `RecordListItem` that refers specifically to an unreleased record version doesn't get a GUID, it only gets a version number. So we have to check to see if the version number is included in the history response.

It works, and I could conceive of a scenario in which it's valid, it just seems odd

Reply:
Reported as a bug in Granta MI, will fix in a later PR if fixed upstream